strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactController.java

```diff
@@ -354,25 +354,22 @@ protected String getRootUri()
                        .collect(Collectors.toList());
     }
 
-    private Predicate createSearchPredicate(String filter,
-                                            String searchTerm)
+    private RepositorySearchRequest createSearchPredicate(String filter,
+                                                          String searchTerm)
     {
-        Predicate rootPredicate = Predicate.empty();
-
-        if (filter != null && !filter.trim().isEmpty())
-        {
-           NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(filter);
-           rootPredicate = t.parseQuery().getPredicate();
-        }
-
-        rootPredicate.and(Predicate.of(ExpOperator.EQ.of("artifactCoordinates.coordinates.extension", "nupkg")));
-
+        String artifactId = null;
         if (searchTerm != null && !searchTerm.trim().isEmpty())
         {
-            rootPredicate.and(Predicate.of(ExpOperator.LIKE.of("artifactCoordinates.coordinates.id",
-                                                               "%" + searchTerm + "%")));
+            artifactId = searchTerm;
+        } else if (filter != null && !filter.trim().isEmpty())
+        {
+           NugetODataFilterQueryParser t = new NugetODataFilterQueryParser(filter);
+           Predicate rootPredicate = t.parseQuery().getPredicate();
+           List<Predicate> predicates = rootPredicate.getChildPredicateList();
+           //TODO: rework NugetODataFilterQueryParser
```

Comment:
What needs to be re-worked? Is this still relevnt? If it is, could you please raise a follow-up task, if you haven't done so already? Thanks! :)

Reply:
#1979